### PR TITLE
Add "rerun_example_" prefix to all our user-visible app-ids

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/2967

The exact prefix is not super important. Once there, we can search-replace easily.

I also added the `app_id_starts_with_rerun_example` analytics signal for this (needed since we hash the app id).
The old `is_official_example` is still there (its not implemented for C++ though).

Review commit-by-commit

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3112) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3112)
- [Docs preview](https://rerun.io/preview/fd985a5a2b561385c581bdf66e8bb1d4fe684ccb/docs) 
- [Examples preview](https://rerun.io/preview/fd985a5a2b561385c581bdf66e8bb1d4fe684ccb/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: docs/code-examples/depth_image_3d.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 image[130:180, 100:280] = 45000
 
 
-rr.init("depth_image", spawn=True)
+rr.init("rerun-example-depth_image", spawn=True)
```
Comment (nikolausWest) [inline docs/code-examples/depth_image_3d.py:12]: I think all these actually look quite ugly. I'd honestly prefer if we just used regular whitespace for all the recording id's. What's even the reason for using snake or kebab case?

Comment (nikolausWest) [reply docs/code-examples/depth_image_3d.py:12]: With "these" I'm particularly referring to the snake_-kebab

**Review by nikolausWest — CHANGES_REQUESTED** [review]
I think we need to decide on how to format the recording id strings and then do it the same everywhere. This looks way too sloppy as one of the first touch points a user gets of Rerun.

I prefer "Rerun example: annotation context rects" but we could also go with "rerun-example-annotation-context-rects" or "rerun_example_annotation_context_rect", as long as we're consistent.

File: docs/code-examples/depth_image_3d.py (hunk 1 above)
Comment (emilk) [reply docs/code-examples/depth_image_3d.py:12]: The reason is that this is an _id_, not a name.

A user reading `"Rerun example: annotation context rects"` is likely to interpret that string as a title, and would probably be surprised that changing it slightly would have side-effects, like Rerun no longer associating a previously stored blueprint with the new recording.

**Comment by emilk** [comment]
I switched all app-ids to `snake_case` now

**Review by nikolausWest — APPROVED** [review]
👍

**Pull request merged**